apps/api/routes/integration.ts

```
@@ -42,4 +42,28 @@ router.post(
   posthogErrorMiddleware('Create Integration')
 );
 
+router.delete(
+  '/:integrationId',
+  posthogMiddleware('Delete Integration'),
+  async (req: Request<{ integrationId: string }>, res: Response<Record<string, never>>) => {
+    const integration = await integrationService.getById(req.params.integrationId, /* unsafe */ true);
+    const { refreshToken } = integration.credentials;
+    const developerConfig = await developerConfigService.getDeveloperConfig();
+
+    // Log out of Salesforce first
+    const oauth2 = new jsforce.OAuth2({
```

Non-blocking: Just for my own curiosity, Is this idempotent? What happens if we call this twice?

---

Do you mean line 59? If we call the logout endpoint twice, the second request *should* fail with a 400 (because the token is invalid). I didn't test logging out twice, but I did test revoking the token twice.

---

I guess what I meant is perhaps if the user is already logged out, we should just skip this step instead of trying to log out (and then throwing an error).

But this is non-blocking.

---

Ah yeah, good point, added a todo for this.